_posts/2019-12-23-newpipe-0.18.0-released.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title: "NewPipe 0.18.0 released, adding PeerTube support"
+short: "NewPipe 0.18.0 released"
+date: 2019-12-13 00:00:00 +0100
+categories: [release, pinned]
+author: TobiGr
+image: peertube # peertube logo
+excerpt_separator: <!-- more -->
+---
+
+Santa came early this year and dropped NewPipe 0.18.0! In this new version support for [PeerTube](https://joinpeertube.org/) has been added and many improvements to the downlaoder were made. Additionally many bugs have been fixed and a couple of cool improvements allow an even smoother usage.
+<!-- more -->
+
+## <img src="/img/peertube-brand.png" style="height:30px; margin-top:-5px;" /> support
+
+[@yausername](https://github.com/yausername) put much effort into adding a fourth service to NewPipe: PeerTube. In case you do not know PeerTube, one can describe it as an open source decentralized video streaming platform.
+
+NewPipe supports most of PeerTubes native features like searching, playing and downloading videos or reading comments. There is one structural point which differs PeerTube from all other so far supported services. Because PeerTube is decentralized, there are [houndreds of instances](https://instances.joinpeertube.org) available in the fediverse. We choose to use [FramaTube](https://framatube.org/) as default instance, because the main organization behind PeerTube is the software developer Framasoft. NewPipe allows you to select the instances you like and add them to your personal list. You can switch between them seamlessly.
+
+------------------ TWO screenshots  ------------------------
+
+We'd like to provide some kind of help to point out donating options for selected instances. We'd like to hear your opinion on this topic and are happy about suggestions where to add donation hints. Please feel free to leave your comment [here](https://github.com/TeamNewPipe/NewPipe/issues/2839).
+Also, PeerTube supports peer-to-peer, but NewPipe does not yet. We'd welcome any help and ideas for a proper implementation. In case you are interested in helping out, you can find the current discussion over at [GitHub](https://github.com/TeamNewPipe/NewPipeExtractor/issues/79).
+
+## Downloader improvements
+
+[@kapadomy](https://github.com/kapadomy) continued his awesome work on the downloader. in this release he calculated an estimated remaining downlaod time which is shown on the download panel. To improve support of more external audio players, audio which is encoded as OPUS is now saved as OGG in a .webem file. In case you needed to pause a download (or some network interruption forced you to do so), NewPipe can now recover the expired download links and resume it after a long time. In case there should be an error while downloading an audio or video, the app now displays more debug info helping to track the cause down.
```

Comment:
```suggestion
[@kapadomy](https://github.com/kapadomy) continued his awesome work on the downloader. in this release he calculated an estimated remaining download time which is shown on the download panel. To improve support of more external audio players, audio which is encoded as OPUS is now saved as OGG instead of a .webm file. In case you needed to pause a download (or some network interruption forced you to do so), NewPipe can now recover the expired download links and resume it after a long time. In case there should be an error while downloading an audio or video, the app now displays more debug info helping to track the cause down.
```

Reply:
it's hard to see in the diff but there are 2 typos in there

Reply:
Spotted em, it's `instead of` and `webem -> webm`. I think `.webm` should even be escaped.

Reply:
```suggestion
[@kapadomy](https://github.com/kapadomy) continued his awesome work on the downloader. in this release he calculated an estimated remaining download time which is shown on the download panel. To improve support of more external audio players, audio which is encoded as Opus is now saved in a `.ogg` file instead of a `.webm` file. In case you needed to pause a download (or some network interruption forced you to do so), NewPipe can now recover the expired download links and resume it after a long time. In case there should be an error while downloading an audio or video, the app now displays more debug info helping to track the cause down.
```

I added escaping to the file extension and and fixed the format names (it's Opus and `.ogg` now (the latter refers to the container in the file only, it's not like any conversion would take place), not OPUS and OGG; I might be a *little* pedantic there).

Reply:
I took a closer at the PR. We switched from downloading Opus audio as `.opus` to `.webm`, but it is planed to switch to `.ogg` in a later PR. That's why I'd remove that line as the change will only be effective for short time.